android/java/org/chromium/chrome/browser/firstrun/WelcomeOnboardingActivity.java

```diff
@@ -251,82 +270,60 @@ private boolean isDefaultBrowser() {
         return BraveSetDefaultBrowserUtils.isBraveSetAsDefaultBrowser(this);
     }
 
-    private void startTimer(int delayMillis) {
-        new Handler().postDelayed(this::nextOnboardingStep, delayMillis);
-    }
-
-    ActivityResultLauncher<String> mRequestPermissionLauncher = registerForActivityResult(
-            new ActivityResultContracts.RequestPermission(), isGranted -> { startTimer(3000); });
+    ActivityResultLauncher<String> mRequestPermissionLauncher =
+            registerForActivityResult(
+                    new ActivityResultContracts.RequestPermission(),
+                    isGranted -> {
+                        nextOnboardingStep();
+                    });
 
     private void nextOnboardingStep() {
         if (isActivityFinishingOrDestroyed()) return;
 
         mCurrentStep++;
+        // Step 0: Handle default browser setup
         if (mCurrentStep == 0) {
-            showIntroPage();
-        } else if (mCurrentStep == 1) {
-            if (!isNewOnboardingEnabled()
-                    || !BraveSetDefaultBrowserUtils.supportsDefaultRoleManager()) {
+            // For devices that don't support role manager API, show browser selection page
+            if (!BraveSetDefaultBrowserUtils.supportsDefaultRoleManager()) {
+                mIvBrave.setVisibility(View.VISIBLE);
                 showBrowserSelectionPage();
-            } else if (!isDefaultBrowser()) {
+            }
+            // If Brave is not default browser, trigger default browser prompt
+            else if (!isDefaultBrowser()) {
                 setDefaultBrowserAndProceedToNextStep();
-            } else {
+            }
+            // If already default browser, proceed to next step
+            else {
                 nextOnboardingStep();
             }
-        } else if (mCurrentStep == getAnalyticsConsentPageStep()) {
+        }
+        // Step 1: Show analytics consent page
+        else if (mCurrentStep == getAnalyticsConsentPageStep()) {
+            mIvBrave.setVisibility(View.VISIBLE);
             showAnalyticsConsentPage();
-        } else {
+        }
+        // Final step: Complete onboarding
+        else {
+            // Set onboarding preferences
             OnboardingPrefManager.getInstance().setP3aOnboardingShown(true);
             OnboardingPrefManager.getInstance().setOnboardingSearchBoxTooltip(true);
+
+            // Mark first run flow as complete
             FirstRunStatus.setFirstRunFlowComplete(true);
+
+            // Accept terms of service and EULA
```

Comment:
I'd remove this comment because it says the same thing as the following method invocations.